x-pack/legacy/plugins/apm/public/components/shared/useDelayedVisibility/index.ts

```diff
@@ -26,6 +26,10 @@ export function useDelayedVisibility(
       setIsVisible(visibility);
     });
     delayedRef.current = delayed;
+
+    return () => {
+      delayed.destroy();
+    };
```

Comment:
Thanks, much better. I wonder if the lack of proper cleanup caused bugs in the past?